### PR TITLE
orchestrator: improvements to the orch host ls command

### PR DESCRIPTION
**Pull request by pcuzner**
This patch provides a --detail 'switch' to orch host ls to
show more host related data for cephadm based clusters.

Fixes: https://tracker.ceph.com/issues/58150

Signed-off-by: Paul Cuzner <pcuzner@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [X] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [X] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [X] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [X] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by pcuzner** [comment]
Code is not complete yet. I need to add support for the cephadm agent case.

**Comment by pcuzner** [comment]
Tested with the agent active, no issues.

**Review by anthonyeleven — APPROVED** [review]
docs LGTM, no code approval should be implied.

**Comment by pcuzner** [comment]
jenkins retest this please

**Comment by pcuzner** [comment]
I've seen the tox errors - will aim to fix in the next day or so.

File: doc/cephadm/host-management.rst [hunk 1]
```diff
@@ -24,6 +24,16 @@ In commands of this form, the arguments "host-pattern", "label" and
 - Any combination of these filtering flags is valid. You may filter against
   name, label and/or status simultaneously.
 
+The "detail" parameter provides more host related information for cephadm based
```
Comment (anthonyeleven) [inline doc/cephadm/host-management.rst:27]: suggest two backticks instead of doublequotes

**Comment by pcuzner** [comment]
jenkins retest this please

**Comment by pcuzner** [comment]
tox is picking up a flake8 issue in the mgr/prometheus module, which is not one of the files in this PR?

**Comment by adk3798** [comment]
jenkins test make check

**Review by adk3798 — APPROVED** [review]
1 point about code location. Otherwise, LGTM

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-01-12_12:11:23-orch:cephadm-wip-adk-testing-2023-01-11-2322-distro-default-smithi/

Quite a few failures, but I think I've pinned all of them down to either a known issue or a failure caused by a specific PR in the run. That being said, we should be able to merge the PRs that are not the two causing the failures or related to cephadm upgrade or ingress service, since those tests were affected by the failures.

11 failed jobs

-  4 mds upgrade sequence jobs failing with `timeout expired in wait_until_healthy`. This was caused by an issue in the prometheus module that, at the time of testing, existed in the latest pacific devel container builds (which those tests use as an upgrade start point). Issue should go away with next pacific devel container build as the fix was merged in pacific already.
- 3 test_nfs task failures `Test failure: test_non_existent_cluster (tasks.cephfs.test_nfs.TestNFS)`. This was caused by one of the PRs in the run adjusting the return code of an nfs module function (not this PR though).
- 2 failed staggered upgrade tests. From what I can tell, the staggered upgrade actually went fine. For whatever reason `ceph versions` json output didn't contain an OSD index with the versions of the OSDs that the test checks.  I ran two instances of the test on main branch and the issue reproduced https://pulpito.ceph.com/adking-2023-01-13_05:56:45-orch:cephadm-main-distro-default-smithi/ 
```
{
    "mon": {
        "ceph version 18.0.0-1762-gcb17f286 (cb17f286272f7ae9dbdf8117ca7b077b0a5cf650) reef (dev)": 3
    },
    "rgw": {
        "ceph version 16.2.0 (0c2054e95bcd9b30fdd908a79ac1d8bbc3394442) pacific (stable)": 2
    },
    "overall": {
        "ceph version 16.2.0 (0c2054e95bcd9b30fdd908a79ac1d8bbc3394442) pacific (stable)": 2,
        "ceph version 18.0.0-1762-gcb17f286 (cb17f286272f7ae9dbdf8117ca7b077b0a5cf650) reef (dev)": 3
    }
}
```
so confident it's not caused by the PRs in the run. Will investigate further later
- 1 instance of https://tracker.ceph.com/issues/57771
- 1 instance of https://tracker.ceph.com/issues/49287

10 Dead Jobs

- 6 failures due to known issue with satellite server during RHEL tests `'The DMI UUID of this host (00000000-0000-0000-0000-0CC47AD93798) matches other registered hosts:`
- 4 failures due to what seems to be mismatch issue between keepalived and haproxy conf, caused by another PR in the run (not this one)

 I think most of the PRs in the run are okay for merging despite the raw failure count.

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by adk3798** [comment]
jenkins test api

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-01-21_05:38:21-orch:cephadm-wip-adk-testing-2023-01-20-1359-distro-default-smithi/

Lots of failures (13) but all accounted for

- 3 instances of `'list' object has no attribute 'get'`. Failure was on new test for setting mon crush location introduced by a PR in the run. That PR can't be merged, but this shouldn't block merging anything else.
- 2 instances of test_cephadm.sh test failing, caused by the same mon crush location PR. Failed with `local variable 'config_json' referenced before assignment`
- 1 instance of https://tracker.ceph.com/issues/49287, known issue
- 2 instances of https://tracker.ceph.com/issues/58524, known issue
- 2 instances of https://tracker.ceph.com/issues/58535, known issue
- 1 instance of https://tracker.ceph.com/issues/57771, known issue
- 1 instance of https://tracker.ceph.com/issues/57755, known issue
- 1 failed basic smoke test that timed out waiting for all 8 OSDs to come up. Did 5 reruns of this test with the same build (https://pulpito.ceph.com/adking-2023-01-22_15:28:20-orch:cephadm-wip-adk-testing-2023-01-20-1359-distro-default-smithi/) all of which passed. My guess is an I/O error creating the last OSD, which I've seen happen (infrequently) when we make a bunch of OSDs on the split up nvme drive we use for having multiple OSDs on these machines. Considering it a 1 off for now, and not blocking merging over it.

Overall, PRs in the run should be okay to merge other than the mon crush location one causing failures. Will start to try and clean up the test suite now that we're able to make builds and run tests again.

**Pull request merged**